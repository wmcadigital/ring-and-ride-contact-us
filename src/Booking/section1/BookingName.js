import { useFormState } from "react-final-form";

import { required, name, composeValidators } from "../../common/validation";
import FormSection from "../../common/FormSection";
import Question from "../../common/Question";
import ProgressIndicator from "../../common/ProgressIndicator";
import TextInput from "../../common/TextInput";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutSectionName from "./getAboutSectionName";

const BookingName = () => {
  const stateApi = useFormState();

  const errorFirstName = stateApi.submitFailed
    ? stateApi.errors?.firstName
    : null;

  const errorLastName = stateApi.submitFailed
    ? stateApi.errors?.lastName
    : null;

  const formValues = stateApi.values;
  const question =
    formValues["bookingParty"] === "behalfSomeone"
      ? "Who are you booking for?"
      : "What is your name?";

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("1", stateApi)}
        sectionName={getAboutSectionName(stateApi)}
      />
      <Question text={question} />
      <TextInput
        fieldName="firstName"
        label="First Name"
        validation={composeValidators(required, name)}
        error={errorFirstName}
      />
      <TextInput
        fieldName="lastName"
        label="Last Name"
        validation={composeValidators(required, name)}
        error={errorLastName}
      />
    </FormSection>
  );
};

export default BookingName;
