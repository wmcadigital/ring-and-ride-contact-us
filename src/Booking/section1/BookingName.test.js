import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import BookingName from "./BookingName";

jest.mock("react-final-form");

describe("BookingName", () => {
  it("individual is filling out form for themselves", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingName />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfSomeone",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingName />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for a group", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfGroup",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingName />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
        firstName: "13545",
      },
      submitFailed: true,
      errors: { firstName: "Invalid", lastName: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingName />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
