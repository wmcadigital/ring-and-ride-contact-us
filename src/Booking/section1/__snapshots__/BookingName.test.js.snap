// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingName individual is filling out form for a group 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 1 of 5"
  />
  <Question
    text="What is your name?"
  />
  <TextInput
    error={null}
    fieldName="firstName"
    label="First Name"
    validation={[Function]}
  />
  <TextInput
    error={null}
    fieldName="lastName"
    label="Last Name"
    validation={[Function]}
  />
</FormSection>
`;

exports[`BookingName individual is filling out form for someone else 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About the applicant"
    sectionPosition="Section 1 of 4"
  />
  <Question
    text="Who are you booking for?"
  />
  <TextInput
    error={null}
    fieldName="firstName"
    label="First Name"
    validation={[Function]}
  />
  <TextInput
    error={null}
    fieldName="lastName"
    label="Last Name"
    validation={[Function]}
  />
</FormSection>
`;

exports[`BookingName individual is filling out form for themselves 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 1 of 4"
  />
  <Question
    text="What is your name?"
  />
  <TextInput
    error={null}
    fieldName="firstName"
    label="First Name"
    validation={[Function]}
  />
  <TextInput
    error={null}
    fieldName="lastName"
    label="Last Name"
    validation={[Function]}
  />
</FormSection>
`;

exports[`BookingName renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 1 of 4"
  />
  <Question
    text="What is your name?"
  />
  <TextInput
    error="Invalid"
    fieldName="firstName"
    label="First Name"
    validation={[Function]}
  />
  <TextInput
    error="Required"
    fieldName="lastName"
    label="Last Name"
    validation={[Function]}
  />
</FormSection>
`;
